joblib/test/test_memory.py

```diff
@@ -217,7 +217,10 @@ def f(x):
                 # MemorizedFunc.__reduce__.
                 Parallel(n_jobs=2)(delayed(cached_f)(i) for i in [1, 2])
                 # Ensure the child process has time to close the file.
-                time.sleep(0.2)
+                for _ in range(5):
```

Comment:
One second is maybe a bit violent, why not keep the 0.2 but increase retries?
```suggestion
                # Wait up to 5 seconds for slow CI runs
                for _ in range(25):
```